flow/connectors/bigquery/merge_stmt_generator.go

```diff
@@ -100,6 +101,32 @@ func (m *mergeStmtGenerator) generateFlattenedCTE() string {
 		m.syncBatchID, m.dstTableName)
 }
 
+// This function is to support datatypes like JSON which cannot be compared by BigQuery
+func (m *mergeStmtGenerator) transformedPkeyStrings(forPartition bool) []string {
+	pkeys := make([]string, 0, len(m.normalizedTableSchema.PrimaryKeyColumns))
+	for _, col := range m.normalizedTableSchema.Columns {
+		if slices.Contains(m.normalizedTableSchema.PrimaryKeyColumns, col.Name) {
```

Comment:
May need to move this into a set if there's a lot of columns & primary keys. For replica identity it'll be `O(N*N)`. But for now should be fine